### ISSUE TITLE
SC3036 false positive on ash (In Dash, echo flags besides -n are not supported)

### ISSUE THREAD

**Issue by felipecrs** [koalaman/shellcheck#3100]
#### For bugs
- Rule Id: SC3036
- My shellcheck version: 0.10.0
- [x] The rule's wiki page does not already cover this (https://shellcheck.net/wiki/SC3036)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=ash

echo -e "\nhi\nthere\nyou"
```

#### Here's what shellcheck currently says:

```
echo -e "\nhi\nthere\nyou"
     ^-- [SC3036](https://www.shellcheck.net/wiki/SC3036) (error): In dash, echo flags besides -n not supported.
```

#### Here's what I wanted or expected to see:

Nothing, as this is supported:

```console
$ docker run --rm -it alpine ash -c 'echo -e "\nhi\nthere\nyou"'

hi
there
you

```

```console
$ docker run --rm -it busybox sh -c 'echo -e "\nhi\nthere\nyou"

hi
there
you

```

**Comment by wileyhy** [comment]
Hm, I haven't used busybox before, AFAIK, but here's how dash behaves in
Termux on Android:

```shell
~ $ bash --version
GNU bash, version 5.2.37(1)-release ...
~ $ dash
$ echo -e '\nfoo'

-e
foo
$ echo '\nfoo'

foo
$
```
